An unfriendly error occurs when the hosted zone does not exist on Route53

## Description

During local setup initialization, an unfriendly error occurs when the host zone on Route53 does not exist.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xc1942d]

goroutine 1 [running]:
github.com/kubefirst/kubefirst/internal/aws.TestHostedZoneLiveness(0xa0?, {0x7ffdb1637dd3, 0x9}, {0x0, 0x0})
        /Users/jdietz/git/github/kubefirst/kubefirst/internal/aws/aws.go:172 +0x7cd
github.com/kubefirst/kubefirst/cmd.glob..func9(0x2daaca0?, {0xc0001b7c20, 0x0, 0xa})
        /Users/jdietz/git/github/kubefirst/kubefirst/cmd/init.go:196 +0x1b7c
github.com/spf13/cobra.(*Command).execute(0x2daaca0, {0xc0001b79a0, 0xa, 0xa})
        /Users/jdietz/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:860 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x2da9b20)
        /Users/jdietz/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:974 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/jdietz/go/pkg/mod/github.com/spf13/cobra@v1.4.0/command.go:902
github.com/kubefirst/kubefirst/cmd.Execute()
        /Users/jdietz/git/github/kubefirst/kubefirst/cmd/root.go:26 +0x25
main.main()
        /Users/jdietz/git/github/kubefirst/kubefirst/main.go:65 +0x41d
```

No additional information is sent in the log.

## How to simulate:

Execute the init command using a AWS account without the hosted zone specified in option `--hosted-zone-name`. Eg:

`kubefirst init --admin-email myemail@email.com --cloud aws --hosted-zone-name nonexistentdomain.com --region us-east-1 --profile default`

--- a/cmd/init.go
+++ b/cmd/init.go
@@ -193,7 +193,10 @@ to quickly create a Cobra application.`,
 		// todo: this doesn't default to testing the dns check
 		skipHostedZoneCheck := viper.GetBool("init.hostedzonecheck.enabled")
 		if !skipHostedZoneCheck {
-			aws.TestHostedZoneLiveness(dryRun, hostedZoneName, hostedZoneId)
+			hostedZoneLiveness := aws.TestHostedZoneLiveness(dryRun, hostedZoneName, hostedZoneId)
+			if !hostedZoneLiveness {
+				log.Panic("Fail to check the Liveness of HostedZone, we need a valid public HostedZone on the same AWS account that Kubefirst will be installed.")
+			}
 		} else {
 			log.Println("skipping hosted zone check")
 		}

--- a/internal/aws/aws.go
+++ b/internal/aws/aws.go
@@ -24,8 +24,8 @@ import (
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/s3"
-	stsV1 "github.com/aws/aws-sdk-go/service/sts"
 	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	stsV1 "github.com/aws/aws-sdk-go/service/sts"
 	"github.com/cip8/autoname"
 	"github.com/google/uuid"
 	"github.com/kubefirst/kubefirst/pkg"
@@ -120,10 +120,10 @@ func GetAccountInfo() {
 	viper.WriteConfig()
 }
 
-func TestHostedZoneLiveness(dryRun bool, hostedZoneName, hostedZoneId string) {
+func TestHostedZoneLiveness(dryRun bool, hostedZoneName, hostedZoneId string) bool {
 	if dryRun {
 		log.Printf("[#99] Dry-run mode, TestHostedZoneLiveness skipped.")
-		return
+		return true
 	}
 	//tracker := progress.Tracker{Message: "testing hosted zone", Total: 25}
 
@@ -168,6 +168,7 @@ func TestHostedZoneLiveness(dryRun bool, hostedZoneName, hostedZoneId string) {
 		})
 		if err != nil {
 			log.Println(err)
+			return false
 		}
 		log.Println("record creation status is ", record.ChangeInfo.Status)
 	} else {
@@ -203,6 +204,7 @@ func TestHostedZoneLiveness(dryRun bool, hostedZoneName, hostedZoneId string) {
 			log.Panicf("unable to resolve hosted zone dns record. please check your domain registrar")
 		}
 	}
+	return true
 	// todo delete route53 record
 
 	// recordDelete, err := route53Client.ChangeResourceRecordSets(context.TODO(), &route53.ChangeResourceRecordSetsInput{

